src/main/java/org/broadinstitute/hellbender/engine/filters/GoodCigarFilter.java

```diff
@@ -0,0 +1,83 @@
+package org.broadinstitute.hellbender.engine.filters;
+
+import htsjdk.samtools.Cigar;
+import htsjdk.samtools.CigarOperator;
+import htsjdk.samtools.SAMRecord;
+
+import java.util.regex.Pattern;
+
+/**
+ * Filter out reads with wonky cigar strings.
+ *
+ *  - No reads with a different length and cigar length
+ *  - No reads with Hard/Soft clips in the middle of the cigar
+ *  - No reads starting with deletions (with or without preceding clips)
+ *  - No reads ending in deletions (with or without follow-up clips)
+ *  - No reads that are fully hard or soft clipped
+ *  - No reads that have consecutive indels in the cigar (II, DD, ID or DI)
+ *
+ * From SAM spec:
+ *  - H can only be present as the first and/or last operation
+ *  - S may only have H operations between them and the ends of the CIGAR string
+ *
+ *  ps: apparently an empty cigar is valid.
+ */
+public final class GoodCigarFilter implements ReadFilter {
+
+    private static final String DIGITS = "\\d*"; //Any operator can be preceeded by a series of digits.
```

Comment:
True. A cigar string without a number is malformed so you can't make a samrecord. In the regexp setting you can at least test the string itself and validate that if there ever were a read with this cigar we would reject it. Anyway, fixed.